### PR TITLE
Report known js libraries.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This is something the amo-validator does so we need that
for compatibility reasons.

Refs mozilla/addons-server#3015

### CODE REVIEW

File: tests/test.linter.js [hunk 1]
```diff
@@ -840,13 +840,16 @@ describe('Linter.extractMetadata()', function() {
...
-      //   'angular.js': 'angularjs.1.2.28.angular.min.js',
-      //   'my/nested/library/path/j.js': 'jquery.2.1.4.jquery-2.1.4.min.js',
-      // });
+      assert.equal(Object.keys(metadata.jsLibs).length, 1);
+      assert.deepEqual(metadata.jsLibs, {
+        'my/nested/library/path/j.js': 'jquery.2.1.4.jquery.min.js',
+      });
+
+      var notices = addonLinter.collector.notices;
+      assert.equal(notices.length, 3);
```
Comment (muffinresearch) [inline tests/test.linter.js]: What are the other notices - are they relevant?

Comment (EnTeQuAk) [inline tests/test.linter.js]: Not to this test, one is "missing manifest.json" for example.

File: docs/rules.md [hunk 2]
```diff
@@ -121,6 +121,7 @@ A :white_check_mark: next to a section of rules means they have all been filed i
 | ----- | ------- | --------- | ---------- | ----------- | --------- | ---------- | -------- | -------- |
 | :white_check_mark: | error | BANNED_LIBRARY | | This version of a JS library is banned for security reasons. | | | | BANNED_LIBRARY |
 | :white_check_mark: | warning | UNADVISED_LIBRARY | | This version of a JS library is not recommended. | | | | UNADVISED_LIBRARY |
+| :white_check_mark: | notice | KNOWN_LIBRARY | This version of a JS library is known and generally accepted. | | | | KNOWN_LIBRARY |
```
Comment (muffinresearch) [inline docs/rules.md]: There's a reference to `blacklisted_js_library` in this doc - does that now match up to what this does? If so the other line should probably be removed.

**Comment by muffinresearch** [comment]
r+wc

File: docs/rules.md (hunk 2 above)
Comment (EnTeQuAk) [inline docs/rules.md]: ah, thanks! I updated `old code` and removed the other line.

**Comment by tofumatt** [comment]
r+ from me too

**Pull request merged**